### ISSUE TITLE
Can we make ani-cli more minimal again?

### ISSUE THREAD

**Issue by ura43** [pystardust/ani-cli#255]
Bold move, I know, but the makefile has definitely caused way more harm than good.
ani-cli is a simple shellscript, that's the beauty of it, symlinking it to /usr/local/bin just for it to be in the path is not worth the countless issues that have popped up and continue to pop up around it.

It was originally meant to make installation easier, but the following issues tell a different tale: #247  #246 #242  #241 #205 #178 #164  #159 

I'm aware that this breaks the current automatic update checking, but that one is janky at best as well and doesn't lead to a good user experience. I like to work in a "move fast, break things" kinda way and would hate to have a user who has no issues using ani-cli update only to have something break for them. Tho a "If you have bugs, update first" disclaimer in the contributing.md wouldn't hurt.

The icing on the cake: we get rid of the make and git dependency

**Comment by ura43** [comment]
@RaynardGerraldo @Dink4n

**Comment by RaynardGerraldo** [comment]
> Bold move, I know, but the makefile has definitely cause way more harm than good.
> ani-cli is a simple shellscript, thats the beauty of it, symlinking it to /usr/local/bin just for it to be in the path is not worth the countless issues that have popped up and continue to pop up around it.
> 
> It was originally meant to make installation easier, but the following issues tell a different tale: #247  #246 #242  #241 #205 #178 #164  #159 
> 
> I'm aware that this breaks the current automatic update checking, but that one is janky at best as well and doesn't lead to a good user experience. I like to work in a "move fast, break things" kinda way and would hate to have a user who has no issues using ani-cli update only to have something break for them. Tho a "If you have bugs, update first" disclaimer in the contributing.md wouldn't hurt.
> 
> The icing on the cake: we get rid of the make and git dependency

I agree, its also not helping considering this is more of a mac bug and ani cli cant even build in m1 chip.

So lets say if the makefile is removed,we'll be running the script with a simple ./ani-cli right?

And for the automatic update checking part,at first i kinda like the feature ,but seeing what you just said about user updating ani-cli only for it to break makes me think that maybe we could keep the feature but make it optional? But then again user would have to find out that it broke for them after they updated so i think its best to just remove it for now at least until we find some sort of solution 

And tbh I havent run makefile with ani-cli since the first fix I made,its so annoying when I thought my fix didnt work and turns out I didnt do sudo make yet,just stopped using it after that and ran it with ./ani-cli😂

**Comment by ura43** [comment]
It's not even just about ani-cli potentially breaking with updates.

Our current implementation sucks on several levels

**Comment by RaynardGerraldo** [comment]
> It's not even just about ani-cli potentially breaking with updates.
> 
> Our current implementation sucks on several levels

Yea☹️ atleast we're slowly cutting down the bug issues

**Comment by 0x8FF** [comment]
I think we should still include the Makefile. It is causing more issues then it should but it’s easier to run ``make`` then having to run multiple commands to have ani-cli in env.

**Comment by tekmyndaspy** [comment]
I agree with you that the symlink is certainly a problem, and that a simple `cp` operation would be best, but I believe removing the Makefile may be a bit hasty, as it is very useful for people who don't want to navigate through a possibly complicated directory structure. As an alternative, it could be cloned to a user local directory, such as `~/.local/bin` rather than `/usr/local`. A bonus to that is that it would get rid of the requirement for `su`, `sudo`, `doas`, or similar commands. That said, it technically is not necessary as one could simply run the script directly already. Maybe it would be best to consider it an optional dependency along the lines of VLC?
\
As for considering `git` to be not a dependency, while you're technically correct I feel it would be overall a bad move to not consider it as such (maybe it could at least be mentioned as an optional dependency) simply due to the nature of the project as something that is constantly updating and is on the move. A user doesn't exactly have any 'releases' to look to, so running a `git pull` is likely the next best option.
\
If one keeps the Makefile, but 'dumbs it down' so to speak, while it would likely be impossible to do automatic update checking, I feel it would certainly be possible to do _update reminding_ where, if the file is a day or so old, it reminds to user that they should run `git pull` to update the cloned repository then run `make update` in order to get the file in the path updated. This allows to keep the simplicity of the script but also allow for people to be reminded to stay relatively up to date.
As an aside, I do know that an update function in the Makefile doesn't exist, but I feel it would be worth it to introduce one (in effect just removing the file in the path then copying it again) as running `cp` on a destination with a file already present can cause some issues.
\
\
**TL;DR:** The symlink should be removed, but in the event it isn't, one should put the symlink in the local path as putting a symlink in a global path can cause trouble with users trying to access other users' home directories, which usually is not allowed. Git, while shouldn't be a strict dependency, should be very encouraged due to the nature of the project as quickly updating. The Makefile could be kept, along with the simplicity of the script as well as keeping people up to date by having the script check for the age of the file, and if it is over a day old reminding people to update.

**Comment by Dink4n** [comment]
How about a versioning system for checking updates?

**Comment by ura43** [comment]
Then we should
- move back from ln to cp
- remove update checking
- add update reminder based on file age
- list git as an optional dependency

I propose a file age of >7 days should give the reminder
A versioning system is a good idea in general but I think we should tackle that one seperatly another time

**Comment by CoolnsX** [comment]
> It's not even just about ani-cli potentially breaking with updates.
> 
> Our current implementation sucks on several levels

I think we should revert back to our old ani-cli and remove all this makefile... And start over again.. like we know how the script is working so can make the old one more optimised and appealing..
Its just a suggestion...

**Comment by ura43** [comment]
> > It's not even just about ani-cli potentially breaking with updates.
> > Our current implementation sucks on several levels
> 
> I think we should revert back to our old ani-cli and remove all this makefile... And start over again.. like we know how the script is working so can make the old one more optimised and appealing.. Its just a suggestion...

So like rewrite the thing from scratch?

**Comment by Dink4n** [comment]
There's an old-ani-cli branch.

**Comment by ura43** [comment]
True, there is. 
Feature creep is certainly something to watch out for, maybe we should open a discussion and see what we see as essential features

**Comment by ura43** [comment]
And take a few things back out of ani-cli

**Comment by Dink4n** [comment]
Shouldn't have merged all those features

**Comment by ura43** [comment]
Yeah, rookie mistake with this being my first major project to manage.
We'll definitely have to reduce our scope again.
Maybe open a "bloated-ani-cli" branch to archive it for those who dislike minimalism for some reason

**Comment by ura43** [comment]
We're at almost 600 lines of code, old-ani-cli is barely above 300

**Comment by ura43** [comment]
Closing in favor of #264

**Comment by CoolnsX** [comment]
What we can do here.. is to make 2 branches... One for full of features.. and one for minimalist... And then we can assign the maintainer for each of the branch... And if a feature is worth putting in both branches.. then we can put it..

We can make another repo instead of branches... If u prefer that way...

**Comment by Dink4n** [comment]
That will be hard to manage

We can continue the discussion in #264
